.github/workflows/build.yml

```diff
@@ -29,7 +38,7 @@ jobs:
         id: npm-cache
         with:
           path: ${{ steps.npm-cache-dir-path.outputs.dir }}
-          key: ${{ runner.os }}-npm-${{ hashFiles('**/package-lock.json') }}
+          key: ${{ runner.os }}-${{ matrix.architecture }}-node${{ matrix.node }}-npm-${{ hashFiles('**/package-lock.json') }}
           restore-keys: |
             ${{ runner.os }}-npm-
```

Comment:
You should make sure that you use a restore key that matches the node version and architecture.

Reply:
Isn't it already the case with $matrix.node and $matrix.archi? Idk much ^^

Reply:
The key is the identify for the cache witch will be created. The restore key is the prefix for all keys witch are candidates for restoring the cache at the begin of a run.

Reply:
Most likely you only want to restore a cache with the "correct" node version and architecture